bin/cardanoCliUtils.js

```diff
@@ -0,0 +1,234 @@
+import { exec } from "child_process";
+import { rimraf } from "rimraf";
+
+const getNet = (net) => {
+  let _net = "";
+  switch (net) {
+    case "devnet":
+      _net = "--testnet-magic=9";
+      break;
+    case "preprod":
+      _net = "--testnet-magic=1";
+      break;
+    case "preview":
+      _net = "--testnet-magic=2";
+      break;
+    case "main":
+    case "mainnet":
+      _net = "--mainnet";
+      break;
+    default:
+      console.log(
+        "Wrong network! Allowed values: devnet, preview, preprod, mainnet",
+        net,
+      );
+      break;
+  }
+  return _net;
+};
+
+const queryUTxOUsingCardanoCli = async (walletAddress, net = "preview") => {
+  return new Promise((resolve, reject) => {
+    exec(
+      `cardano-cli query utxo \
+        --address ${walletAddress} \
+        ${getNet(net)}`,
+      (error, stdout, stderr) => {
+        if (error) {
+          reject(error.message);
+          return;
+        }
+        if (stderr) {
+          reject(stderr);
+          return;
+        }
+        const _match = stdout.match(/([a-z0-9]{64}) *(\d) *(\d+)/);
+        // console.log(">stdout:", stdout, _match)
+        if (_match === null) {
+          resolve({ TxHash: 0, TxIx: 0, Amount: 0 });
+        } else {
+          const TxHash = _match[1];
+          const TxIx = parseInt(_match[2]);
+          const Amount = parseInt(_match[3]);
+
+          resolve({ TxHash, TxIx, Amount });
+        }
+      },
+    );
+  });
+};
+
+// **********************************************************************************************************
+const createDraftTransaction = (
+  walletAddress,
+  metadataFilePath,
+  TxHash,
+  TxIx,
+) => {
+  return new Promise((resolve, reject) => {
+    const cmd = `cardano-cli transaction build-raw \
+                    --tx-in ${TxHash}#${TxIx} \
+                    --tx-out ${walletAddress}+0 \
+                    --metadata-json-file ${metadataFilePath} \
+                    --fee 0 \
+                    --out-file tx.draft`;
+
+    exec(cmd, (error, stdout, stderr) => {
+      if (error) {
```

Comment:
I would write this in less lines. For instance you don't need return statements. Use if/else instead